### PR TITLE
fix: added type rename support for enum, union and interface.

### PR DESCRIPTION
**Pull request by laststylebender14**
**Summary:**  
previously rename types transformer supported only renaming in normal types, now extended the support to enum, union and interface types.

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/config/transformer/rename_types.rs [hunk 1]
```diff
@@ -65,6 +83,51 @@ impl Transform for RenameTypes {
                         }
                     }
                 }
+
+                // replace in interface.
+                type_.implements = type_
+                    .implements
+                    .iter()
+                    .filter_map(|interface_type_name| {
```
Comment (karatakis) [inline src/core/config/transformer/rename_types.rs:91]: Why are you using `filter_map` when you are always returning `Some`? 

Can a normal `map` with `match` work instead?

File: src/core/config/transformer/rename_types.rs [hunk 2]
```diff
@@ -65,6 +83,51 @@ impl Transform for RenameTypes {
...
+                        lookup
+                            .get(interface_type_name)
+                            .cloned()
+                            .or(Some(interface_type_name.clone()))
```
Comment (karatakis) [inline src/core/config/transformer/rename_types.rs:95]: This is eagerly evaluated, use `or_else`

File: src/core/config/transformer/rename_types.rs [hunk 3]
```diff
@@ -65,6 +83,51 @@ impl Transform for RenameTypes {
...
+                    union_type_.types.insert(type_name);
+                }
+            }
+
+            // replace in union as well.
+            for union_type_ in config.unions.values_mut() {
+                union_type_.types = union_type_
+                    .types
+                    .iter()
+                    .filter_map(|type_name| {
```
Comment (karatakis) [inline src/core/config/transformer/rename_types.rs:127]: 1) why we also use `filter_map`?
2) replace `or` with `or_else` because it lazily evaluates

**Review by karatakis — COMMENTED** [review]
I found some slips, but overall its a great work.

File: src/core/config/transformer/rename_types.rs (hunk 1 above)
Comment (laststylebender14) [reply src/core/config/transformer/rename_types.rs:91]: yep, replace with map.

File: src/core/config/transformer/rename_types.rs (hunk 2 above)
Comment (laststylebender14) [reply src/core/config/transformer/rename_types.rs:95]: using `unwrap_or`.

**Review by karatakis — APPROVED** [review]

**Pull request merged**